### ISSUE TITLE
Option to use executeLargeUpdate

### ISSUE THREAD

**Issue by chrisknoll** [OHDSI/DatabaseConnector#249]
I'm investigating an issue where we are running a large scale CohortIncidence sql via JDBC, and have run into a 'update count value is out of range'.

This occurs when we perform an insert/update statement (in this case, most likely an insert) that exceeds MAX_INT number of rows (about 2Bil).

I tried to find in code where we call executeSql, and found [this reference](https://github.com/OHDSI/DatabaseConnector/blob/c2ae089af5e0be8eb9de4b9e2c39e539b9ed695e/R/DBI.R#L447).  note where it expects the update result to be a java.lang.Integer result.

In the MS Sql JDC driver, it performs a [check ](https://github.com/microsoft/mssql-jdbc/blob/145c10e90148087e48c6c18f0f3449bcbb3d1708/src/main/java/com/microsoft/sqlserver/jdbc/SQLServerStatement.java#L757)to see if update resulted in > MAXINT.

So, considering we may be dealing with large scale data, we may want to use executeLargeUpdate, described [here](https://docs.oracle.com/javase/8/docs/api/java/sql/Statement.html#executeLargeUpdate).

I don't think we should change the behavior of executeSql, but rather expose a executeLargeSql() that will call out to executeLargeBatch, vs how it currently calls executeBatch as seen [here](https://github.com/OHDSI/DatabaseConnector/blob/c2ae089af5e0be8eb9de4b9e2c39e539b9ed695e/R/Sql.R#L444).

**Comment by schuemie** [comment]
The DBI [specifies](https://dbi.r-dbi.org/reference/dbsendstatement) that `dbSendStatement()` should return a `DBIResult` object that can be used to retrieve the number of affected rows. We currently get that number by calling JDBC's `getUpdateCount()` [here](https://github.com/OHDSI/DatabaseConnector/blob/c2ae089af5e0be8eb9de4b9e2c39e539b9ed695e/R/Sql.R#L314C52-L314C66). Oddly enough, the [JDBC specifications](https://docs.oracle.com/javase/8/docs/api/java/sql/Statement.html) themselves state that `getUpdateCount()` returns an int. 

So in other words, I think switching to `executeLargeUpdate()` won't be enough. We also need some way to get the number of affected rows when that number exceeds what can fit in an int. Thoughts?

**Comment by chrisknoll** [comment]
> We currently get that number by calling JDBC's getUpdateCount() [here](https://github.com/OHDSI/DatabaseConnector/blob/c2ae089af5e0be8eb9de4b9e2c39e539b9ed695e/R/Sql.R#L314C52-L314C66). Oddly enough, the [JDBC specifications](https://docs.oracle.com/javase/8/docs/api/java/sql/Statement.html) themselves state that getUpdateCount() returns an int.

That's why we have [getLargeUpdateCount()](https://docs.oracle.com/javase/8/docs/api/java/sql/Statement.html#getLargeUpdateCount--).

I don't think we should change anything along the standard dbSendStatement() path.   Except maybing providnng a way to specify an `option()` or optional param that will make dbSendStatement() go down the `executeLargeUpdate()`, `executeLargeBatch()`, `getlargeUpdateCount()` path.

Eventually, we are going to have to deal with analytic calls that is going to exceed the number of rows affected > MAX_INT.

**Comment by schuemie** [comment]
Ah, `getLargeUpdateCount()` would work yes. 

What would be the downside to just always using `executeLargeUpdate()` and `getLargeUpdateCount()`?

**Comment by schuemie** [comment]
I've implemented a first version of this in the `large_update` branch. Seems to work fine on most platforms, but the DataBricks JDBC driver throws an error when using `executeLargeUpdate()` for some queries so DataBricks still uses `execute()`.

**Comment by chrisknoll** [comment]
Excellent!   I'll ask my collaborator to pull that branch and execute the analysis again to see if that avoids the 'value too large' error.

-Chris

**Comment by chrisknoll** [comment]
Sorry I didn't report back, but the collaborator reported success. Thank you!
